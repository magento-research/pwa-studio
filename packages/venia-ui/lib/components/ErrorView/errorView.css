.root {
    padding: 1rem;
}

.content {
    max-width: 50%;
    margin: auto;
    display: grid;
    grid-gap: 1rem;
    text-align: center;
}

.header {
    font-size: var(--venia-global-typography-heading-L-fontSize);
    font-weight: var(--venia-global-fontWeight-semibold);
}

.message {
}

.actionsContainer {
}

@media (max-width: 640px) {
    .content {
        max-width: 100%;
    }
}
