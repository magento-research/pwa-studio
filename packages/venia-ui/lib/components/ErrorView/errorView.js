import React from 'react';
import { fullPageLoadingIndicator } from '../LoadingIndicator';
import { string } from 'prop-types';

export const ERROR_TYPES = {
    LOADING: 'loading',
    NOT_FOUND: 'notFound'
};

export const errorMap = new Map()
    .set('default', 'Something went wrong. Please try again.')
    .set('loading', fullPageLoadingIndicator)
    .set('notFound', 'That page could not be found. Please try again.');

const ErrorView = props => {
    const { type } = props;
    const key = type || 'default';

    return <h1>{errorMap.get(key)}</h1>;
};

ErrorView.propTypes = {
    type: string
};

export default ErrorView;
