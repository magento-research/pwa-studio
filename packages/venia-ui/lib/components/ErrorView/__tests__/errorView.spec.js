import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import ErrorView, { ERROR_TYPES, errorMap } from '../errorView';

const renderer = new ShallowRenderer();

test('it renders the correct tree when loading', () => {
    const tree = renderer.render(<ErrorView type={ERROR_TYPES.LOADING} />);

    expect(tree).toMatchSnapshot();
});

test('it renders the correct tree when page not found', () => {
    const tree = renderer.render(<ErrorView type={ERROR_TYPES.NOT_FOUND} />);

    expect(tree).toMatchSnapshot();
});

test('it renders the correct tree when out of stock', () => {
    ERROR_TYPES.CUSTOM = 'custom';
    errorMap.set(ERROR_TYPES.CUSTOM, 'Nice Error Message!');
    const tree = renderer.render(<ErrorView type={ERROR_TYPES.CUSTOM} />);

    expect(tree).toMatchSnapshot();
});

test('it renders the internal error tree otherwise', () => {
    const tree = renderer.render(<ErrorView />);

    expect(tree).toMatchSnapshot();
});
