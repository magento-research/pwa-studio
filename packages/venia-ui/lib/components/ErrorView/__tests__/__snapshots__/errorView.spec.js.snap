// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the correct tree when loading 1`] = `
<h1>
  <LoadingIndicator
    global={true}
  >
    Fetching Data...
  </LoadingIndicator>
</h1>
`;

exports[`it renders the correct tree when out of stock 1`] = `
<h1>
  Nice Error Message!
</h1>
`;

exports[`it renders the correct tree when page not found 1`] = `
<h1>
  That page could not be found. Please try again.
</h1>
`;

exports[`it renders the internal error tree otherwise 1`] = `
<h1>
  Something went wrong. Please try again.
</h1>
`;
